Add support for JSON format

erocarrera opened the pull request:
This small change adds support to read Stan's JSON data in addition to Rdump. This was requested in #1466 and I've needed it myself. It updates `_read_data()` to parse as JSON any input files with `.json` extension, defaulting to Rdump otherwise.
Hence `observed_data`, `constant_data`, and `predictions_constant_data` all accept JSON inputs now.
I've run `black` and `pylint` on the changes.

Also added a missing bracket in a docstring.

ahartikainen commented on arviz/data/io_cmdstan.py:
```diff
@@ -1,5 +1,6 @@
 # pylint: disable=too-many-lines
 """CmdStan-specific conversion code."""
+import json
```
```suggestion
try:
    import ujson as json
except ImportError:
    import json
```

erocarrera replied:
Oh yes! that's great!

ahartikainen reviewed (commented):
This is a great addition.

We have ujson as an optional dependency so I think we could always try to import it first

ahartikainen commented:
Great.

Should we add some test for this? something similar that we have for rdump?

erocarrera commented:
> Great.
> 
> Should we add some test for this? something similar that we have for rdump?

Most definitely. I've added a json version of one of the Rdump files in tests and updated the tests to make sure both pass meet them same assertions.
